sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -1008,27 +1008,6 @@ def DeletePermission(
         permission_id = base.GetResourceIdOrFullNameFromLink(permission_link)
         return self.DeleteResource(path, "permissions", permission_id, None, options, **kwargs)
 
-    def ReadItems(
-        self,
-        collection_link: str,
-        feed_options: Optional[Mapping[str, Any]] = None,
-        response_hook: Optional[Callable[[Mapping[str, Any], Mapping[str, Any]], None]] = None,
-        **kwargs: Any
-    ) -> ItemPaged[Dict[str, Any]]:
-        """Reads all documents in a collection.
-
-        :param str collection_link: The link to the document collection.
-        :param dict feed_options: The additional options for the operation.
-        :param response_hook: A callable invoked with the response metadata.
-        :type response_hook: Callable[[Dict[str, str], Dict[str, Any]]
-        :return: Query Iterable of Documents.
-        :rtype: query_iterable.QueryIterable
-
-        """
-        if feed_options is None:
-            feed_options = {}
-
-        return self.QueryItems(collection_link, None, feed_options, response_hook=response_hook, **kwargs)
 
     def QueryItems(
```

Comment:
The fetch_function in the query method below is currently acting on `self.last_response_headers`. This might also need to be adjusted?

Reply:
This would probably be fixed by updating `def __QueryFeed` to return `Tuple[List[Dict[str, Any]], Dict[str, Any]` rather than using `fetch_fn` to build this tuple (i.e. we would return the result of `__QueryFeed` directly.